src/lib/accessibility/FocusZone/focusUtilities.ts

```diff
@@ -399,3 +400,46 @@ export function isElementFocusSubZone(element?: HTMLElement): boolean {
     element.getAttribute(FOCUSZONE_SUB_ATTRIBUTE) === 'true'
   )
 }
+
+let targetToFocusOnNextRepaint: HTMLElement | { focus: () => void } | null | undefined = undefined
+
+/**
+ * Sets focus to an element asynchronously. The focus will be set at the next browser repaint,
+ * meaning it won't cause any extra recalculations. If more than one focusAsync is called during one frame,
+ * only the latest called focusAsync element will actually be focused
+ * @param element The element to focus
+ */
+export function focusAsync(element: HTMLElement | { focus: () => void } | undefined | null): void {
+  if (element) {
+    // An element was already queued to be focused, so replace that one with the new element
+    if (targetToFocusOnNextRepaint) {
+      targetToFocusOnNextRepaint = element
+      return
+    }
+
+    targetToFocusOnNextRepaint = element
+
+    const win = getWindow(element as Element)
+
+    if (win) {
+      // element.focus() is a no-op if the element is no longer in the DOM, meaning this is always safe
+      win.requestAnimationFrame(() => {
+        targetToFocusOnNextRepaint && targetToFocusOnNextRepaint.focus()
+
+        // We are done focusing for this frame, so reset the queued focus element
+        targetToFocusOnNextRepaint = undefined
+      })
+    }
+  }
+}
+
+/**
+ * Helper to get the window object.
+ *
+ * @public
+ */
+export function getWindow(rootElement?: Element | null): Window | undefined {
+  return rootElement && rootElement.ownerDocument && rootElement.ownerDocument.defaultView
```

Comment:
could be simplified: `return rootElement && rootElement.ownerDocument && rootElement.ownerDocument.defaultView || window`